src/core/psxhw.cc

```diff
@@ -371,6 +371,20 @@ uint32_t PCSX::HW::read32(uint32_t add) {
             hard = g_emulator->m_mem->msanRealloc(ptr, size);
             break;
         }
+        case 0x1f802094: {
+            uint32_t headerAddr = g_emulator->m_cpu->m_regs.GPR.n.a0;
+            if (headerAddr == 0x1f802094) {
+                g_system->printf(_("Recursive msanGetChainPtr\n"));
```

Comment:
On one hand, this error message might need to be a bit more explicit, like "pcsx_msanGetChainPtr called with invalid header pointer". On the other, it'd be really weird for this one to actually happen. But I see the idea, yes. It might be worth using the `PSXAddress` class to validate the pointer more generally speaking? It's invalid to try and do a DMA on anything other than RAM that is, as you can't DMA from the scratchpad or the hardware registers that is. Same with `pcsx_msanSetChainPtr` below.